File: 프로그래머스/unrated/120808. 분수의 덧셈/분수의 덧셈.js
function catBox(number1, number2) {
    if (number1 % number2 === 0) return number2;
    else {
        return catBox(number2, number1 % number2);
    }
}

function tongbun(numer1, denom1, numer2, denom2) {
    const numer = numer1 * denom2 + numer2 * denom1;
    const denom = denom1 * denom2;
    
    return [numer, denom]
}

function yacbun(numer, denom) {
    const divisor = catBox(numer,denom);
    
    numer /= divisor;
    denom /= divisor;
    
    return [numer, denom];
}



function solution(numer1, denom1, numer2, denom2) {
    let answer = [];
    
    let [numer, denom] = tongbun(numer1, denom1, numer2, denom2);

    [numer, denom] = yacbun(numer, denom);
    
    answer = [numer, denom];

    return answer;
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
def solution(n1, d1, n2, d2):
    
    for i in range(99999999999999999999):
        if int(n1 * (i+1) / d1  + n2 * (i+1)/ d2) == n1 * (i+1) / d1 + n2 * (i+1) / d2:

            return [int(n1 * (i+1) / d1 + n2 * (i+1) / d2), (i+1)]